docs/adr/0024-source-level-edit.md

```diff
@@ -0,0 +1,274 @@
+An example is provided below as an illustration.
+
+Additional capabilities could be built on top of the combination of TAMS and OTIO, for example rendering an OTIO composition using lightweight Flow copies (and new objects for the transitions as in the examples above), or generating OTIO as part of a metadata-driven editorial workflow.
+
+* Good, because it allows for complex compositions in a consistent way.
+* Good, because it moves a significant amount of complexity into an existing technology.
+* Good, because OTIO has growing support in other tools (e.g. NLEs).
+* Good, because the flexible plugin model in OTIO (e.g. Media Linkers) could be used to bridge into other tools: for example fetching Flows as a file locally for an NLE without direct TAMS support.
+* Neutral, because it requires incorporating an additional tech stack.
+* Bad, because OTIO might be overkill for simple operations such as basic clipping.
+
+## Appendix: OpenTimelineIO TAMS References
+
+This appendix illustrates how TAMS references might work in OpenTimelineIO compositions.
+
+### URL Form
+
+This form uses an `ExternalReference` to a URL in a TAMS instance.
+
+Notice that the URL has a prefix `tamss://` (for "TAMS Secure" - `tams://` would also work for HTTP).
+In addition the `start_time` and `duration` in the `available_range` are Flow timestamps with nanosecond precision,
+referring to the timerange over which the Flow is available.
+The `source_range` also has a `start_time` as a nanosecond timestamp within the same Flow.
+
+```json
+{
+    "OTIO_SCHEMA": "Clip.2",
+    "metadata": {},
+    "name": "camera-one.ts",
+    "source_range": {
+        "OTIO_SCHEMA": "TimeRange.1",
+        "duration": {
+            "OTIO_SCHEMA": "RationalTime.1",
+            "rate": 50.0,
+            "value": 111.0
+        },
+        "start_time": {
+            "OTIO_SCHEMA": "RationalTime.1",
+            "rate": 1000000000.0,
+            "value": 1723124225400000000.0
+        }
+    },
+    "effects": [],
+    "markers": [],
+    "enabled": true,
+    "media_references": {
+        "DEFAULT_MEDIA": {
+            "OTIO_SCHEMA": "ExternalReference.1",
+            "metadata": {},
+            "name": "camera-one.ts",
+            "available_range": {
+                "OTIO_SCHEMA": "TimeRange.1",
+                "duration": {
+                    "OTIO_SCHEMA": "RationalTime.1",
+                    "rate": 1000000000.0,
+                    "value": 372100000000.0
+                },
+                "start_time": {
+                    "OTIO_SCHEMA": "RationalTime.1",
+                    "rate": 1000000000.0,
+                    "value": 1723124086620000000
+                }
+            },
+            "available_image_bounds": null,
+            "target_url": "tamss://tams.example.com/flows/9bb414a5-862c-494f-86ce-8e2720ecc315"
+        }
+    },
+    "active_media_reference_key": "DEFAULT_MEDIA"
+}
+```
+
+### Reference Form
```

Comment:
This extends the URL form with metadata? Or should the `target_url` be changed to a TAMS base URL?

Reply:
I've made this a `MissingReference` in 1b305cf and expanded the text to better explain why the `metadata` is also useful